mqtt-streaming/src/test/java/docs/javadsl/MqttFlowTest.java

```diff
@@ -0,0 +1,219 @@
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertTrue;
+
+public class MqttFlowTest {
+
+  private static ActorSystem system;
+  private static Materializer materializer;
+
+  private static Pair<ActorSystem, Materializer> setupMaterializer() {
+    final ActorSystem system = ActorSystem.create("MqttFlowTest");
+    final Materializer materializer = ActorMaterializer.create(system);
+    return Pair.create(system, materializer);
+  }
+
+  @BeforeClass
+  public static void setup() {
+    final Pair<ActorSystem, Materializer> sysmat = setupMaterializer();
+    system = sysmat.first();
+    materializer = sysmat.second();
+  }
+
+  @AfterClass
+  public static void teardown() {
+    TestKit.shutdownActorSystem(system);
+  }
+
+  @Test
+  public void establishClientBidirectionalConnectionAndSubscribeToATopic()
+      throws InterruptedException, ExecutionException, TimeoutException {
+    String clientId = "flow-spec/flow";
+    String topic = "source-spec/topic1";
+
+    // #create-streaming-flow
+    MqttSessionSettings settings = MqttSessionSettings.create();
+    MqttClientSession session = new ActorMqttClientSession(settings, materializer, system);
+
+    Flow<ByteString, ByteString, CompletionStage<Tcp.OutgoingConnection>> connection =
+        Tcp.get(system).outgoingConnection("localhost", 1883);
+
+    Flow<Command<?>, DecodeErrorOrEvent, NotUsed> mqttFlow =
+        Mqtt.clientSessionFlow(session).join(connection);
+    // #create-streaming-flow
+
+    // #run-streaming-flow
+    Pair<SourceQueueWithComplete<Command<?>>, CompletionStage<DecodeErrorOrEvent>> run =
+        Source.<Command<?>>queue(3, OverflowStrategy.fail())
+            .via(mqttFlow)
+            .drop(3)
+            .toMat(Sink.head(), Keep.both())
+            .run(materializer);
+
+    SourceQueueWithComplete<Command<?>> commands = run.first();
+    commands.offer(new Command(new Connect(clientId, ConnectFlags.None())));
+    commands.offer(new Command(new Subscribe(topic)));
+    commands.offer(new Command(new Publish(topic, ByteString.fromString("ohi"))));
+    // #run-streaming-flow
+
+    CompletionStage<DecodeErrorOrEvent> event = run.second();
+    DecodeErrorOrEvent decodeErrorOrEvent = event.toCompletableFuture().get(3, TimeUnit.SECONDS);
+    assertTrue(decodeErrorOrEvent.getEvent().isPresent());
+  }
+
+  @Test
+  public void establishServerBidirectionalConnectionAndSubscribeToATopic()
+      throws InterruptedException, ExecutionException, TimeoutException {
+    String clientId = "flow-spec/flow";
+    String topic = "source-spec/topic1";
+    String host = "localhost";
+    int port = 9884;
+
+    // #create-streaming-bind-flow
+    MqttSessionSettings settings = MqttSessionSettings.create();
+    MqttServerSession session = new ActorMqttServerSession(settings, materializer, system);
```

Comment:
I wonder if you should have a `create` method in `MqttServerSession` instead? 
Many are puzzled by this kind of distinction (as we have with `Materializer` and `ActorMaterializer`).

Reply:
Any comment to this?